contracts/treasury/GovernorCountingOverridable.sol

```diff
@@ -0,0 +1,229 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.9;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/governance/extensions/GovernorVotesQuorumFractionUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/interfaces/IERC5805Upgradeable.sol";
+
+import "../bonding/libraries/EarningsPool.sol";
+import "../bonding/libraries/EarningsPoolLIP36.sol";
+
+import "../Manager.sol";
+import "../IController.sol";
+import "../rounds/IRoundsManager.sol";
+import "../bonding/IBondingCheckpoints.sol";
+
+interface IVotes is IERC5805Upgradeable {
+    function delegatedAt(address account, uint256 timepoint) external returns (address);
+}
+
+/**
+ * @title GovernorCountingOverridable
+ * @notice Implements the Counting module from OpenZeppelin Governor with support for delegators overriding their
+ * delegated transcoder's vote. This module is used through inheritance by the Governor contract.
+ */
+abstract contract GovernorCountingOverridable is Initializable, GovernorUpgradeable {
+    using SafeMath for uint256;
+
+    error InvalidVoteType(uint8 voteType);
+    error VoteAlreadyCast();
+
+    function __GovernorCountingOverridable_init() internal onlyInitializing {
+        __GovernorCountingOverridable_init_unchained();
+    }
+
+    function __GovernorCountingOverridable_init_unchained() internal onlyInitializing {}
+
+    /**
+     * @dev Supported vote types. Matches Governor Bravo ordering.
+     */
+    enum VoteType {
+        Against,
+        For,
+        Abstain
+    }
+
+    /**
+     * @dev Tracks state of specicic voters in a single proposal.
+     */
+    struct ProposalVoterState {
+        bool hasVoted;
+        VoteType support;
+        // This vote deductions state is only necessary to support the case where a delegator might vote before their
+        // transcoder. When that happens, we need to deduct the delegator(s) votes before tallying the transcoder vote.
+        uint256 deductions;
+    }
+
+    /**
+     * @dev Tracks the tallying state for a proposal vote counting logic.
+     */
+    struct ProposalTally {
+        uint256 againstVotes;
+        uint256 forVotes;
+        uint256 abstainVotes;
+        mapping(address => ProposalVoterState) voters;
+    }
+
+    mapping(uint256 => ProposalTally) private _proposalTallies;
+
+    /**
+     * @dev See {IGovernor-COUNTING_MODE}.
+     */
+    // solhint-disable-next-line func-name-mixedcase
+    function COUNTING_MODE() public pure virtual override returns (string memory) {
+        return "support=bravo&quorum=for,abstain,against";
```

Comment:
Why does this include against, but [GovernorCountingSimpleUpgradeable](https://github.com/OpenZeppelin/openzeppelin-contracts-upgradeable/blob/3d4c0d5741b131c231e558d7a6213392ab3672a5/contracts/governance/extensions/GovernorCountingSimpleUpgradeable.sol#L43) does not?

Reply:
2 main reasons:
- Our [current poll tallying logic](https://github.com/livepeer/explorer/blob/c6c7cb1b8cc5fab66c575dd63f03a53aab87c2b1/lib/api/polls.ts#L76) considers both "yes" and "no" votes as participation. So I guess we'd also need to rethink our quorum/quota values if we were to change this.
- My personal opinion, as from every source I searched for the term "quorum" in the abstract sense, it means "the amount of people in the room". IMHO anyone voting is in the room and so should be considered for quroum. I found the `GovernorCountingSimple` logic a bit counterintuitive (and even more the bravo impl which only considers `for` votes!)

But if it wasn't for our current voting system I'd pick the consistency with OZ as well, even if it seems weird on a first glance. After all our quorum logic will be different from most Governors out there if we implement that differently here.

Let me know if you'd prefer to keep compatibility with OZ instead of our current polls, or anything else. Don't have a strong opinion about this anyway.

Reply:
Yeah exactly! So the `support=bravo` indicates our association of vote type as a `uint8` to what they represent for the vote (0 against, 1 for, 2 abstain), and the second `quorum` param indicates what we use internally for the quorum. Both of which UIs like Tally _should_ (😆) use to display vote counting info for a proposal.